### PR TITLE
fix: fixes isLoggedIn check to include expiry

### LINKED ISSUE #2420: Failing login shows "session active" message, when it isn't
## Describe the bug

1. Let your session expire. 

2. run `./infra login`
fail login to see the following:
`...Your session as admin to localhost:8443 is still active.`

but they are not logged in. Running `./infra info` shows the correct message:
`...Session expired; run .... to start a new session`

## Environment

```
$ infra version
```

```
$ kubectl version
```

### PR DESCRIPTION
**Pull request by kimskimchi**
## Checklist

- [ ] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #2420

### CODE REVIEW

File: internal/cmd/logout_test.go [hunk 1]
```diff
@@ -7,8 +7,10 @@ import (
 	"path/filepath"
 	"sync/atomic"
 	"testing"
+	"time"
 
 	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/infrahq/infra/api"
```
Comment (mxyng) [inline internal/cmd/logout_test.go:13]: First party imports should be in its own section after the 3rd party imports

Comment (kimskimchi) [reply internal/cmd/logout_test.go]: fixed!

File: internal/cmd/logout_test.go [hunk 2]
```diff
@@ -316,4 +331,12 @@ func TestLogout(t *testing.T) {
 		assert.ErrorContains(t, err, `"infra logout" accepts at most 1 argument`)
 		assert.ErrorContains(t, err, `Usage:  infra logout [SERVER]`)
 	})
+
+	t.Run("expired session fails", func(t *testing.T) {
```
Comment (mxyng) [inline internal/cmd/logout_test.go:335]: this should check if the expired config is removed

File: internal/cmd/logout_test.go [hunk 3]
```diff
@@ -117,6 +122,16 @@ func TestLogout(t *testing.T) {
...
 					Current:       true,
+					Expires:       api.Time(time.Now().Add(time.Hour * 2).UTC().Truncate(time.Second)),
+				},
+				{
+					Name:          "user2",
+					Host:          srv.Listener.Addr().String(),
+					AccessKey:     "the-access-key",
+					UserID:        2,
+					SkipTLSVerify: true,
+					Current:       true,
```
Comment (dnephin) [inline internal/cmd/logout_test.go:133]: I don't think logout calls `mustBeLoggedIn`, so the test case won't cover the change you are making.

**Review by dnephin — COMMENTED** [review]
Fix looks good, but I'm pretty sure the test is not covering this case, because we should expect an error with `Session expired`, but the test case is asserting `NilError`.

I think we can do something like this to test it, maybe in a new test case `TestMustBeLoggedIn`:

```go
cfg := ClientConfig{
    ClientConfigVersion: clientConfigVersion,
    Hosts: []ClientHostConfig{
        // a ClientHostConfig like the "user2" case you added 
     }
}

err := writeConfig(&cfg)
assert.NilError(t, err)

err := Run(context.Background(), "destinations", "list")
assert.ErrorContains(t, err, "Session expired")
```

File: internal/cmd/logout_test.go (hunk 3 above)
Comment (kimskimchi) [reply internal/cmd/logout_test.go:133]: logout calls mustbeLoggedIn, and only prints out a success message if logged in, and nothing if logged out (or expired). 

The test case `expired session fails` has two sessions - one is logged in but api call fails (so api count +1), the other is expired so it won't even call the api (hence api count stays as 1).

File: internal/cmd/logout_test.go (hunk 2 above)
Comment (kimskimchi) [reply internal/cmd/logout_test.go:335]: done (at least I think that is what you meant for me to write up, the equateEmpty() parts).

**Review by dnephin — APPROVED** [review]
Looks good!

**Pull request merged**